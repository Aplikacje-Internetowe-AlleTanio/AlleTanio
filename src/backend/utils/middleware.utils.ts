import { NextFunction, Request, Response } from 'express'
import { verifyToken } from './jwt.utils'
import { ReasonPhrases, StatusCodes } from 'http-status-codes'
import jwt, { JwtPayload } from 'jsonwebtoken'

const SECRET = (process.env.TOKEN_SECRET as string) ?? 'XYZ'

interface CustomRequest extends Request {
    userId?: string
}

export const authorize = (
    req: CustomRequest,
    res: Response,
    next: NextFunction,
) => {
    const token = req.headers.authorization
    const parsedToken = token?.replace('Bearer ', '')
    const result = verifyToken(parsedToken ?? '', SECRET)

    if (!token || !result.isValid) {
        res.send(StatusCodes.UNAUTHORIZED).json({
            errors: [ReasonPhrases.UNAUTHORIZED],
        })
    } else {
        const decodedToken = jwt.decode(token) as JwtPayload
        req.userId = decodedToken.id as string
        next()
    }
}
