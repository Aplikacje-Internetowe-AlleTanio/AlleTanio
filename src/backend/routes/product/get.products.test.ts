import { Response, Request } from 'express';
import { StatusCodes } from 'http-status-codes';
import { prisma } from '../../database';
import { handleRequest, TRequestData } from '../../utils/request.utils';
import getProductsRoute from './get.products';

describe('get.products', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    afterAll(async () => {
        await prisma.$disconnect();
    });

    it('should handle the request and return products based on filters', async () => {
        const mockRequest = {
            query: {
                name: 'parufki',
                fastDelivery: 'true',
                price: '2137',
            },
        } as unknown as Request;
        const mockResponse = {
            status: jest.fn().mockReturnThis(),
            json: jest.fn(),
        } as unknown as Response;

        const requestData: TRequestData<void> = {
            req: mockRequest,
            res: mockResponse,
            execute: async () => getProductsRoute.handler(mockRequest, mockResponse),
            responseSuccessStatus: StatusCodes.OK,
        };

        await handleRequest(requestData);

        expect(mockResponse.status).toHaveBeenCalledWith(StatusCodes.OK);
        expect(mockResponse.json).toMatchSnapshot();
    });
});
