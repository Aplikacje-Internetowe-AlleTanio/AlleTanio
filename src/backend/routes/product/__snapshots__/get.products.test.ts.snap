// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get.products should handle the request and return products based on filters 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": undefined,
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
